pkg/gatewayserver/io/udp/udp.go

```diff
@@ -299,8 +299,9 @@ func (s *srv) handleUp(ctx context.Context, state *state, packet encoding.Packet
 			}
 		}
 		var rtt *time.Duration
-		if cids, delta, ok := state.tokens.Get(uint16(packet.Token[0])<<8|uint16(packet.Token[1]), packet.ReceivedAt); ok {
-			msg.TxAcknowledgment.CorrelationIDs = cids
+		if downlink, delta, ok := state.tokens.Get(uint16(packet.Token[0])<<8|uint16(packet.Token[1]), packet.ReceivedAt); ok {
+			msg.TxAcknowledgment.DownlinkMessage = downlink
+			msg.TxAcknowledgment.CorrelationIDs = downlink.CorrelationIDs
```

Comment:
What is the added value of `CorrelationIDs` over reading `DownlinkMessage.CorrelationIDs`?

Reply:
We must also set `CorrelationIDs` to avoid breaking the API.